packages/core/src/App/Containers/Routes/routes.jsx

```diff
@@ -70,10 +69,12 @@ const Routes = ({
     // non-supported language, the language still
     // shows up in the URL. This is not in sync
     // with the default language (EN), so we
-    // will remove it.
-    if ((!has_lang && lang !== 'EN') || (has_lang && lang === 'EN')) {
-        window.history.replaceState({}, document.title, urlForLanguage(lang));
-    }
+    // will remove it. (The same thing for dark_mode)
+    window.history.replaceState(
+        {},
+        document.title,
+        urlSetQuery({ lang: lang !== 'EN' ? lang : '', dark: is_dark_mode_on ? '1' : '' })
```

Comment:
```suggestion
        urlSetQuery({ lang: lang.replace('EN', ''), dark: Number(is_dark_mode_on)})
```
@sergei-deriv Can we do like this? 🤔

Reply:
@farzin-deriv looks better, agree with you :)